### PR TITLE
Filter categories

### PR DESCRIPTION
**Pull request by san2heg**

### CODE REVIEW

**Comment by tanzeelak** [comment]
Gracias. Could you merge master into your branch? And maybe add a default "All" categories option?

**Comment by tanzeelak** [comment]
If possible, also make your changes in the mapbox files (same way you did with the sidebar) so that they work at the same time

File: src/app/sidebar/sidebar.component.ts [hunk 1]
```diff
@@ -21,24 +26,41 @@ export class SidebarComponent implements OnInit {
...
+      this.selectedEvent = event;
     }
+
+    filterByCategory(category: string): void {
+      if (category === "all") {
+        this.filteredEvents = this.events;
+        return;
+      }
+      var tempEvents = [];
+      for (let e of this.events) {
```
Comment (codyleyhan) [inline src/app/sidebar/sidebar.component.ts]: could also use this.events.filter and replace this entire function

File: src/app/sidebar/sidebar.component.ts [hunk 2]
```diff
@@ -21,24 +26,41 @@ export class SidebarComponent implements OnInit {
...
-        this.selectedEvent = event;
+      this.selectedEvent = event;
     }
+
+    filterByCategory(category: string): void {
+      if (category === "all") {
+        this.filteredEvents = this.events;
+        return;
+      }
+      var tempEvents = [];
```
Comment (codyleyhan) [inline src/app/sidebar/sidebar.component.ts]: If you are using es6 why use nonintuitive variable declarations, const and let are blocked scoped and are the new standard for all js dev

Comment (tanzeelak) [reply src/app/sidebar/sidebar.component.ts]: We were having problems with variable declarations working with "let" on mobile friendly, so Nikhil suggested we only use let.

Comment (codyleyhan) [reply src/app/sidebar/sidebar.component.ts]: There is a let in the for loop?

Comment (tanzeelak) [reply src/app/sidebar/sidebar.component.ts]: Ah, our mistake.

Comment (codyleyhan) [reply src/app/sidebar/sidebar.component.ts]: Also es6 let and const have full browser support http://kangax.github.io/compat-table/es6/

**Comment by san2heg** [comment]
I made a new event service that basically stores state of the current events for the current date and then broadcasts the event to every component subscriber whenever the date changes or events are filtered. this way, events will only ever be fetched once and every component will be updated automatically if events ever change. Don't need map.service anymore. @tanzeelak @jsfuentes lmk what you think

**Pull request merged**